src/lib/homestore_backend/replication_message.hpp

```diff
@@ -57,6 +80,49 @@ struct ReplicationMessageHeader {
     }
 };
 
+struct SyncMessageHeader : public BaseMessageHeader<SyncMessageHeader> {
+    SyncMessageHeader() : BaseMessageHeader() {
+        magic_num = HOMEOBJECT_RESYNC_MAGIC;
+        protocol_version = HOMEOBJECT_RESYNC_PROTOCOL_VERSION_V1;
+    }
+    SyncMessageType msg_type;
+    uint8_t reserved_pad[6]{};
+
+    bool corrupted() const{
+        if (magic_num != HOMEOBJECT_RESYNC_MAGIC || protocol_version != HOMEOBJECT_RESYNC_PROTOCOL_VERSION_V1) {
+            return true;
+        }
+        return BaseMessageHeader::corrupted();
+    }
+
+    std::string to_string() const {
+        return fmt::format(
+            "magic={:#x} version={} msg_type={} payload_size={} payload_crc={} header_crc={}\n",
+            magic_num, protocol_version, enum_name(msg_type), payload_size, payload_crc, header_crc);
+    }
+};
 #pragma pack()
 
+// objId is the logical offset of the snapshot in baseline resync
+// obj_id (64 bits) = type_bit (1 bit) | shard_id (48 bits) | batch_id (15 bits)
+// type_bit = 0 for HomeStore, 1 for HomeObject
+struct objId {
+    snp_obj_id_t value;
+    shard_id_t shardId;
+    snp_batch_id_t batchId;
```

Comment:
why we dont what type?

Reply:
Because if type=0, it won't be passed to HO. When it is used in HO, `type` always=1, so I think no need to add a `type`.